postgresql/resource_postgresql_role.go

```diff
@@ -58,10 +61,27 @@ func resourcePostgreSQLRole() *schema.Resource {
 				Description: "The name of the role",
 			},
 			rolePasswordAttr: {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Sensitive:   true,
-				Description: "Sets the role's password",
+				Type:          schema.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				ConflictsWith: []string{rolePasswordWOAttr, rolePasswordWOVersionAttr},
+				Description:   "Sets the role's password",
+			},
+			rolePasswordWOAttr: {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				ConflictsWith: []string{rolePasswordAttr},
+				RequiredWith:  []string{rolePasswordWOVersionAttr},
+				WriteOnly:     true,
+				Description:   "Sets the role's password without storing it in the state file.",
+			},
+			rolePasswordWOVersionAttr: {
+				Type:          schema.TypeString,
```

Comment:
Should we use an integer like in [Terraform example](https://developer.hashicorp.com/terraform/language/resources/ephemeral/write-only#update-write-only-arguments-with-versions) ?

Reply:
functionally it doesn't matter, my reasoning was that it allows a user to use things like githash variables as a trigger.

example: for my rds databases I update the master password on every pipeline run, I would prefer to use my githash variable I inject from the pipeline this, but as its not an integer I have to do this instead

```
resource "time_static" "trigger" {
  triggers = {
    # The timestamp() function is re-evaluated on every run,
    # ensuring this trigger value always changes.
    always_run = timestamp()
  }
}

master_password_wo_version = time_static.trigger.unix

``` 

Though there is also value in keeping the pattern the same as other projects that use a wo_version

Up to you, its your project :)

Reply:
@cyrilgdn  I'm leaning towards changing it to a integer to keep it in line with other resources doing the same thing. Happy with that?

Reply:
@haydn-j-evans No it's ok, it's a good reasoning for me 👍